client/src/components/HistoryViewer/HistoryViewer.js

```diff
@@ -308,97 +254,111 @@ class HistoryViewer extends Component {
    *
    * @returns {HistoryViewerVersionList|null}
    */
-  renderComparisonSelectionList() {
-    const { compare: { versionFrom }, ListComponent } = this.props;
-
+  const renderComparisonSelectionList = () => {
+    const versionFrom = compare.versionFrom;
     if (!versionFrom) {
       return null;
     }
-
     const selectionListClasses = classNames(
       'history-viewer__table',
       'history-viewer__table--comparison-selected',
     );
-
     return (
       <ListComponent
         versions={[versionFrom]}
         extraClass={selectionListClasses}
       />
     );
-  }
+  };
 
   /**
    * Renders a list of versions
    *
    * @returns {HistoryViewerVersionList}
    */
-  renderVersionList() {
-    const {
-      isInGridField,
-      ListComponent,
-      CompareWarningComponent,
-      compare,
-      compare: { versionFrom: hasVersionFrom },
-    } = this.props;
-
+  const renderVersionList = () => {
+    const hasVersionFrom = compare.versionFrom;
     return (
-      <div className={this.getContainerClasses()}>
+      <div className={getContainerClasses()}>
         <CompareWarningComponent />
 
         <div className={isInGridField ? '' : 'panel panel--padded panel--scrollable'}>
-          {this.renderComparisonSelectionList()}
+          {renderComparisonSelectionList()}
           <ListComponent
-            versions={this.state.versions}
+            versions={versions}
             showHeader={!compare || (compare && !hasVersionFrom)}
-            compareModeAvailable={this.compareModeAvailable()}
+            compareModeAvailable={compareModeAvailable()}
           />
 
           <div className="history-viewer__pagination">
-            {this.renderPagination()}
+            {renderPagination()}
           </div>
         </div>
       </div>
     );
-  }
-
-  renderCompareMode() {
-    const { compare } = this.props;
+  };
 
+  const renderCompareMode = () => {
     if (compare && compare.versionFrom && compare.versionTo) {
-      return this.renderVersionDetail();
+      return renderVersionDetail();
     }
-    return this.renderVersionList();
-  }
+    return renderVersionList();
+  };
 
-  render() {
-    const { graphQLErrors, loading, compare, currentVersion, recordId } = this.props;
+  /**
+   * Refresh version data on initial mount
+   */
+  useEffect(() => {
+    refreshVersionData();
+    // Reset selected version on unmount
```

Comment:
```suggestion
    // Reset selected version on unmount to prevent data leaking between instances
```
Adding back the context from the original comment

Reply:
Updated